src/Functions/searchAnyAll.h

```diff
@@ -2,24 +2,30 @@
 
 #include <Functions/IFunction.h>
 #include <Interpreters/Context_fwd.h>
-#include <Interpreters/GinFilter.h>
 #include <Interpreters/ITokenExtractor.h>
+#include <absl/container/flat_hash_map.h>
 
 namespace DB
 {
 
+enum class SearchAnyAllMode : uint8_t
```

Comment:
I think this should be renamed to `TextSearchMode` as it is in [MergeTreeIndexConditionText.h](https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/MergeTree/MergeTreeIndexConditionText.h#L12). Then I believe the enum in that file can be removed and made to include this file (the cpp already includes it).

Reply:
Fair enough